src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -27,17 +27,20 @@ if(WITH_SYSTEM_NPM)
 else()
   set(mgr-dashboard-nodeenv-dir ${CMAKE_CURRENT_BINARY_DIR}/node-env)
   set(nodeenv NODEENV)
+  set(MGR-DASHBOARD-USERCONFIG --userconfig ${mgr-dashboard-nodeenv-dir}/.npmrc)
 
   add_custom_command(
     OUTPUT "${mgr-dashboard-nodeenv-dir}/bin/npm"
     COMMAND ${CMAKE_SOURCE_DIR}/src/tools/setup-virtualenv.sh --python=${MGR_PYTHON_EXECUTABLE} ${mgr-dashboard-nodeenv-dir}
     COMMAND ${mgr-dashboard-nodeenv-dir}/bin/pip install nodeenv
     COMMAND ${mgr-dashboard-nodeenv-dir}/bin/nodeenv -p --node=12.16.2
+    COMMAND mkdir ${mgr-dashboard-nodeenv-dir}/.npm
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     COMMENT "dashboard nodeenv is being installed"
   )
   add_custom_target(mgr-dashboard-nodeenv
-    COMMAND . ${mgr-dashboard-nodeenv-dir}/bin/activate && npm config set python ${MGR_PYTHON_EXECUTABLE} && deactivate
+    COMMAND . ${mgr-dashboard-nodeenv-dir}/bin/activate && npm config set python ${MGR_PYTHON_EXECUTABLE} ${MGR-DASHBOARD-USERCONFIG} && deactivate
+    COMMAND . ${mgr-dashboard-nodeenv-dir}/bin/activate && npm config set cache ${mgr-dashboard-nodeenv-dir}/.npm ${MGR-DASHBOARD-USERCONFIG} && deactivate
```

Comment:
nit, since we have two command which are supposed to run within the mgr-dashboard-nodeenv, probably we could deactivate later, like
```cmake
add_custom_target(mgr-dashboard-nodeenv
  COMMAND . ${mgr-dashboard-nodeenv-dir}/bin/activate
  COMMAND npm config set python ${MGR_PYTHON_EXECUTABLE} ${MGR-DASHBOARD-USERCONFIG}
  COMMAND npm config set cache ${mgr-dashboard-nodeenv-dir}/.npm ${MGR-DASHBOARD-USERCONFIG} && deactivate
  COMMAND deactivate
  ...
)
```

Reply:
ack

Reply:
Hmmm... seems it doesn't work that way Kefu?

```
cd /home/brad/working/src/ceph-fedora-32-brad/build/src/pybind/mgr/dashboard && mkdir /home/brad/working/src/ceph-fedora-32-brad/build/src/pybind/mgr/dashboard/node-env/.npm
cd /home/brad/working/src/ceph-fedora-32-brad/build/src/pybind/mgr/dashboard && . /home/brad/working/src/ceph-fedora-32-brad/build/src/pybind/mgr/dashboard/node-env/bin/activate
cd /home/brad/working/src/ceph-fedora-32-brad/build/src/pybind/mgr/dashboard && npm config set python /usr/bin/python3.8 --userconfig /home/brad/working/src/ceph-fedora-32-brad/build/src/pybind/mgr/dashboard/node-env/.npmrc
/bin/sh: npm: command not found
```

Switching back to the original.

Reply:
modified it so it's slightly more compact.